### PR TITLE
Add pipeline overrides, add pipeline and installation flags, support China

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/17919, introduces the `pipeline` and `installation` arguments to some commands, and introduces a map where configuration can be overridden for selected pipelines.

The goal is to distinguish between the `provider` of a release and the target installation for a test. This is primarily to support testing `aws` releases in China without establishing China as a separate provider.

The `create release` and `testoperatorrelease` commands already determined the provider based on the diff of the releases diff or passed flag. Those commands now accept a `pipeline` flag to indicate the name of the Tekton Pipeline where standup is running. This name is checked against a list of overrides. If an override for that pipeline name is configured, it will be used instead of the default behavior.

In addition, the usage of the `provider` flag has been aligned. Previously, it was used in some places to refer to the conceptual provider in a Giant Swarm platform sense -- namely to distinguish between releases for AWS, Azure, or KVM. In other places, it was used as the key for identifying which management cluster to use for a test, and which kubeconfig to use to contact that target MC. This second usage was the problematic one and is semantically different from the first usage. In places where the `provider` was used to select a target MC/kubeconfig, those usages have been replaced with `installation` flags.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by stone-z** [comment]
Tested with https://github.com/giantswarm/test-infra/pull/125 on https://github.com/giantswarm/releases/pull/754

File: cmd/cleanup/flag.go [hunk 1]
```diff
@@ -6,26 +6,26 @@ import (
...
+	Installation string
+	ReleaseID    string
 }
 
 func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVarP(&f.ClusterID, flagClusterID, "c", "", `The ID of the cluster to delete.`)
 	cmd.Flags().StringVarP(&f.Config, flagConfig, "g", "", `The path to the file containing API endpoints and tokens for each provider.`)
 	cmd.Flags().StringVarP(&f.Kubeconfig, flagKubeconfig, "k", "", `The path to the directory containing the kubeconfigs for provider control planes.`)
-	cmd.Flags().StringVarP(&f.Provider, flagProvider, "p", "", `The provider of the cluster to delete.`)
+	cmd.Flags().StringVarP(&f.Installation, flagInstallation, "p", "", `The target management cluster type to be used ('aws', 'azure', 'kvm', or 'aws-china').`)
```
Comment (MarcelMue) [inline cmd/cleanup/flag.go:28]:
```suggestion
	cmd.Flags().StringVarP(&f.Installation, flagInstallation, "i", "", `The target management cluster type to be used ('aws', 'azure', 'kvm', or 'aws-china').`)
```

File: cmd/create/cluster/flag.go [hunk 2]
```diff
@@ -9,26 +9,26 @@ import (
...
+	Output       string
+	Release      string
 }
 
 func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVarP(&f.Config, flagConfig, "g", "", `The path to the file containing API endpoints and tokens for each provider.`)
 	cmd.Flags().StringVarP(&f.Kubeconfig, flagKubeconfig, "k", "", `The path to the directory containing the kubeconfigs for provider control planes.`)
 	cmd.Flags().StringVar(&f.Output, flagOutput, "", `The directory in which to store the cluster ID, kubeconfig, and provider of the created cluster.`)
-	cmd.Flags().StringVarP(&f.Provider, flagProvider, "p", "", `The provider of the target control plane.`)
+	cmd.Flags().StringVarP(&f.Installation, flagInstallation, "p", "", `The target management cluster type to be used ('aws', 'azure', 'kvm', or 'aws-china').`)
```
Comment (MarcelMue) [inline cmd/create/cluster/flag.go:31]:
```suggestion
	cmd.Flags().StringVarP(&f.Installation, flagInstallation, "i", "", `The target management cluster type to be used ('aws', 'azure', 'kvm', or 'aws-china').`)
```

**Review by MarcelMue — COMMENTED** [review]
I get it - the `installation` naming is still a bit confusing without context but I can't think of something better.

**Comment by stone-z** [comment]
> I get it - the `installation` naming is still a bit confusing without context but I can't think of something better.

Yeah, I considered other options like `target`, it's just a little awkward because it actually refers to the index/filename from the predetermined list of configs to use and not actually the installation name. Open to suggestions though, it is a bit confusing

**Review by MarcelMue — APPROVED** [review]
Let's try this for now and see how it goes.

**Pull request merged**